tests/repo/init.c

```diff
@@ -247,6 +247,19 @@ void test_repo_init__detect_ignorecase(void)
 		"core.ignorecase", found_without_match ? true : GIT_ENOTFOUND);
 }
 
+void test_repo_init__detect_symlinks(void)
+{
+	struct stat st;
+	bool no_symlinks;
+
+	no_symlinks = (p_symlink("target", "link") < 0 ||
+	    p_lstat("link", &st) < 0 ||
+	    ! (S_ISLNK(st.st_mode)));
```

Comment:
It might make sense to have a common helper function to detect symlink support which unifies these three lines here and in "checkout::index".

Reply:
Agreed and done.